crates/polars-sql/src/context.rs

```diff
@@ -704,38 +730,58 @@ impl SQLContext {
 
         // Remove the group_by keys as polars adds those implicitly.
         let mut aggregation_projection = Vec::with_capacity(projections.len());
-        let mut aliases: BTreeSet<&str> = BTreeSet::new();
+        let mut projection_aliases = PlHashSet::new();
+        let mut group_key_aliases = PlHashSet::new();
 
         for mut e in projections {
```

Comment:
1. This iterates over a `&[Expr]` slice, thus you get a `&Expr` on every iteration. E.g. we borrow this expression from somewhere (the slice).